### PR TITLE
Add Rafter Katacoda tutorials

### PR DESCRIPTION
**Pull request by mmitoraj**
**Description**

Changes proposed in this pull request:

- Added Katacoda tutorials to Rafter

**Related issue(s)**
See also kyma-project/kyma/issues/6292

### CODE REVIEW

File: docs/katacoda-tutorials/step3.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+   metadata:
+     name: content
+     namespace: default
+   spec:
+     region: "us-east-1"
+     policy: readonly
+   EOF
+   ```{{execute}}
+
+3. Apply an Asset CR that points to a single Markdown file. To extract metadata from the Markdown file before it goes to storage, in the Asset CR specify communication with **metadataWebhookService**. The webhook service is a separate service responsible for extracting metadata from Markdown files. As a result, the status of the Asset resource will be enriched with the file metadata. Run:
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/katacoda-tutorials/step3.md]: I think we can skip this part if we explain it at the beginning of the tutorial as I suggested: `The webhook service is a separate service responsible for extracting metadata from Markdown files.`

File: docs/katacoda-tutorials/step4.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+       mutationWebhookService:
+         - name: rafter-rafter-asyncapi-service
+           namespace: default
+           endpoint: "/v1/convert"
+     bucketRef:
+       name: content
+   EOF
+   ```{{execute}}
+
+3. Check if the Asset status is `Ready` which means that fetching and communication with the AsyncAPI Service is completed. Run:
```
Comment (kazydek) [inline docs/katacoda-tutorials/step4.md]:
```suggestion
3. Check if the Asset status is `Ready` which means that fetching and communication with the AsyncAPI Service was completed. Run:
```

File: docs/katacoda-tutorials/step4.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+   `kubectl get assets asyncapi-file -o jsonpath='{.status.phase}'`{{execute}}
+
+To make sure that the file is in storage and you can extract it, proceed with the next two steps:
```
Comment (kazydek) [inline docs/katacoda-tutorials/step4.md]:
```suggestion
To make sure that the file is in storage and you can extract it, follow these steps:
```

File: docs/katacoda-tutorials/setup.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+helm init --service-account=tiller
+
+echo "Waiting for Tiller to be up..."
+LIMIT=30
+COUNTER=0
+
+
+while [ ${COUNTER} -lt ${LIMIT} ] && [ -z "$TILLER_STATUS" ]; do
+  (( COUNTER++ ))
+  echo "Tiller is almost ready, $LIMIT checks left and so far it is attempt number $COUNTER"
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/katacoda-tutorials/setup.sh]: just removed the unnecessary quotation mark
```suggestion
echo "Tiller is almost ready, there are $LIMIT checks left, and it is the $COUNTER attempt so far"
```

File: docs/katacoda-tutorials/setup.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+tar -xzvf helm-"v2.16.0"-linux-amd64.tar.gz
+mv ./linux-amd64/{helm,tiller} /usr/local/bin
+
+echo "Waiting for Minikube to be up..."
+LIMIT=50
+COUNTER=0
+
+while [ ${COUNTER} -lt ${LIMIT} ] && [ -z "$MINIKUBE_STATUS" ]; do
+  (( COUNTER++ ))
+  echo "Minikube is almost ready, $LIMIT checks left and so far it is attempt number $COUNTER"
```
[the remark this answers is not in the document]
Comment (mmitoraj) [reply docs/katacoda-tutorials/setup.sh]: just removed the unnecessary quotation mark and added it at the end of the sentence
```suggestion
  echo "Minikube is almost ready, there are $LIMIT checks left, and it is the $COUNTER attempt so far"
```

File: docs/katacoda-tutorials/basic/step2.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
+In this scenario, you will learn how to use Rafter to store static webpages. You will create a Bucket, push an Asset to it, and open the website from sources stored in the Bucket. For the purpose of this scenario, the Asset is a package containing all static files needed to build a website, such as HTML, JS, and CSS files. Follow these steps:
+
+
+1. Export a URL to ready-to-use sources of a simple website as an environment variable:
+
+   `export GH_WEBPAGE_URL=https://github.com/kyma-project/examples/archive/master.zip`{{execute}}
+
+2. Create a Bucket by applying a Bucket custom resource (CR). Run:
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step2.md]:
```suggestion
2. Create a bucket by applying a Bucket custom resource (CR). Run:
```

File: docs/katacoda-tutorials/basic/step2.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+In this scenario, you will learn how to use Rafter to store static webpages. You will create a Bucket, push an Asset to it, and open the website from sources stored in the Bucket. For the purpose of this scenario, the Asset is a package containing all static files needed to build a website, such as HTML, JS, and CSS files. Follow these steps:
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step2.md]:
```suggestion
In this scenario, you will learn how to use Rafter to store static webpages. You will create a Bucket, push an Asset to it, and open the website from sources stored in the Bucket. For the purpose of this scenario, the asset is a package containing all static files needed to build a website, such as HTML, JS, and CSS files. Follow these steps:
```

File: docs/katacoda-tutorials/basic/step2.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+   metadata:
+     name: pages
+     namespace: default
+   spec:
+     region: "us-east-1"
+     policy: readonly
+   EOF
+   ```{{execute}}
+
+3. Create an Asset by applying an Asset CR. The Rafter Controller Manager fetches the Asset from the location provided in **spec.source.url**. In this example, you can see that the fetched item is a package with a specific directory filtered.
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step2.md]:
```suggestion
3. Create an asset by applying an Asset CR. The Rafter Controller Manager fetches the asset from the location provided in **spec.source.url**. In this example, you can see that the fetched item is a package with a specific directory filtered.
```

File: docs/katacoda-tutorials/basic/step2.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+     source:
+       url: ${GH_WEBPAGE_URL}
+       mode: package
+       filter: /asset-store/webpage/
+     bucketRef:
+       name: pages
+   EOF
+   ```{{execute}}
+
+4. Make sure that the Asset status is `Ready` which means that fetching, unpacking, and filtering was completed. Run:
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step2.md]:
```suggestion
4. Make sure that the status of the asset is `Ready` which means that fetching, unpacking, and filtering was completed. Run:
```

File: docs/katacoda-tutorials/basic/step2.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+   `kubectl get assets webpage -o jsonpath='{.status.phase}'`{{execute}}
+
+5. Export the name of the remote Bucket in storage as an environment variable. This name is available in the Bucket CR status and is not exactly the same as the name of a specific Bucket CR:
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step2.md]:
```suggestion
5. Export the name of the remote bucket in storage as an environment variable. This name is available in the Bucket CR status and is not exactly the same as the name of a specific Bucket CR:
```

File: docs/katacoda-tutorials/basic/step2.md (hunk 9 above)
Comment (kazydek) [reply docs/katacoda-tutorials/basic/step2.md]: (to be consistent with the same step in the next tutorial)

File: docs/katacoda-tutorials/basic/step3.md [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+       metadataWebhookService:
+         - name: rafter-rafter-front-matter-service
+           namespace: default
+           endpoint: "/v1/extract"
+     bucketRef:
+       name: content
+   EOF
+   ```{{execute}}
+
+4. Make sure that the status of the Asset CR is `Ready` which means that fetching and communication with the Front Matter Service was completed. Run:
```
Comment (kazydek) [inline docs/katacoda-tutorials/basic/step3.md]:
```suggestion
4. Make sure that the status of the asset is `Ready` which means that fetching and communication with the Front Matter Service was completed. Run:
```

**Pull request merged**